PWGMM/UE/Tasks/ue-zdc-analysys.cxx

```diff
@@ -0,0 +1,294 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+///
+/// \brief Task for ZDC
+/// \author
+/// \since
+#include "Framework/AnalysisTask.h"
+#include "Framework/AnalysisDataModel.h"
+#include "Framework/HistogramRegistry.h"
+
+#include "Common/DataModel/EventSelection.h"
+//#include "Common/CCDB/EventSelectionParams.h"
+//#include "Common/CCDB/TriggerAliases.h"
+#include "CCDB/BasicCCDBManager.h"
+#include "Common/DataModel/Centrality.h"
+#include "Common/DataModel/Multiplicity.h"
+
+#include "TH1F.h"
+#include "TH2F.h"
+#include "TObjArray.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+void customize(std::vector<o2::framework::ConfigParamSpec>& workflowOptions)
```

Comment:
Please note, that these options cannot be used on hyperloop. The new approach is to use multiple process functions with switches to enable/disable them, that can be used on hyperloop.

Reply:
I am not sure I fully get you point, Anton. Which options are you referring to? Those in defineDataProcessing?
Thanks a lot for this input (there are several things I am not aware of, evidently)

Reply:
Yes, the customize function introduces options that can only be provided via command line, they are not supported on hyperloop, which means you would not be able to configure your task there. Instead of having several tasks that are controlled by these options, you can instead have a single task with multiple process functions that are controlled by the process switches defined in the task itself. These switches can be changed through configuration JSON, which is how the tasks are configured on hyperloop. 

We have an example for this: https://github.com/AliceO2Group/O2Physics/blob/master/Tutorials/src/multiProcess.cxx

Reply:
@aalkin 
I updated my task following your suggestions and I tested it locally.